test/unit/prometheus-servicemonitor.bats

```diff
@@ -0,0 +1,118 @@
+      . || echo "---") | tee /dev/stderr |
+      yq 'length > 0' | tee /dev/stderr)
+  [ "${actual}" = "false" ]
+}
+
+
+@test "prometheus/ServiceMonitor-server: assertEnabled" {
+  cd `chart_dir`
+  local actual=$( (helm template \
+      --show-only templates/prometheus-servicemonitor.yaml  \
+      --set 'telemetry.serviceMonitor.enabled=true' \
+      . || echo "---") | tee /dev/stderr |
+      yq 'length > 0' | tee /dev/stderr)
+  [ "${actual}" = "true" ]
+}
+
+@test "prometheus/ServiceMonitor-server: assertScrapeTimeout default" {
+  cd `chart_dir`
+  local actual=$( (helm template \
+      --show-only templates/prometheus-servicemonitor.yaml \
+      --set 'telemetry.serviceMonitor.enabled=true' \
+      . ) | tee /dev/stderr |
+      yq -r '.spec.endpoints[0].scrapeTimeout' | tee /dev/stderr)
+  [ "${actual}" = "10s" ]
+}
+
+@test "prometheus/ServiceMonitor-server: assertScrapeTimeout update" {
+  cd `chart_dir`
+  local actual=$( (helm template \
+      --show-only templates/prometheus-servicemonitor.yaml \
+      --set 'telemetry.serviceMonitor.enabled=true' \
+      --set 'telemetry.serviceMonitor.scrapeTimeout=60s' \
+      . ) | tee /dev/stderr |
+      yq -r '.spec.endpoints[0].scrapeTimeout' | tee /dev/stderr)
+  [ "${actual}" = "60s" ]
+}
+
+@test "prometheus/ServiceMonitor-server: assertInterval default" {
+  cd `chart_dir`
+  local actual=$( (helm template \
+      --show-only templates/prometheus-servicemonitor.yaml \
+      --set 'telemetry.serviceMonitor.enabled=true' \
+      . ) | tee /dev/stderr |
+      yq -r '.spec.endpoints[0].interval' | tee /dev/stderr)
+  [ "${actual}" = "30s" ]
+}
+
+@test "prometheus/ServiceMonitor-server: assertInterval update" {
+  cd `chart_dir`
+  local actual=$( (helm template \
+      --show-only templates/prometheus-servicemonitor.yaml \
+      --set 'telemetry.serviceMonitor.enabled=true' \
+      --set 'telemetry.serviceMonitor.interval=60s' \
+      . )  | tee /dev/stderr |
+      yq -r '.spec.endpoints[0].interval' | tee /dev/stderr)
+  [ "${actual}" = "60s" ]
+}
+
+@test "prometheus/ServiceMonitor-server: assertSelectors default" {
+  cd `chart_dir`
+  local output=$( (helm template \
+      --show-only templates/prometheus-servicemonitor.yaml \
+      --set 'telemetry.serviceMonitor.enabled=true' \
+      . ) | tee /dev/stderr)
+
+  [ "$(echo "$output" | yq -r '.metadata.labels.control-plane')" = "controller-manager" ]
+}
+
+@test "prometheus/ServiceMonitor-server: assertSelectors override" {
+  cd `chart_dir`
+  local output=$( (helm template \
+      --show-only templates/prometheus-servicemonitor.yaml \
+      --set 'telemetry.serviceMonitor.enabled=true' \
+      --set 'telemetry.serviceMonitor.selectors.baz=qux' \
+      --set 'telemetry.serviceMonitor.selectors.bar=foo' \
+      . ) | tee /dev/stderr)
+
+  [ "$(echo "$output" | yq -r '.metadata.labels | has("app")')" = "false" ]
+  [ "$(echo "$output" | yq -r '.metadata.labels.baz')" = "qux" ]
```

Comment:
not sure if it would be helpful, but you might want to consider checking the length of `.metadata.labels`